packages/foam-vscode/static/graphs/default/graph.js

```diff
@@ -244,11 +252,15 @@ function getLinkColor(link, model) {
   const style = model.style;
   switch (getLinkState(link, model)) {
     case 'regular':
-      return d3.hsl(style.node.note).darker(2);
+      return (
+        style.lineColor || d3.hsl(style.node['note']).copy({ opacity: 0.8 })
+      );
     case 'highlighted':
       return style.highlightedForeground;
     case 'lessened':
-      return d3.hsl(style.node.note).darker(4);
+      return style.lineColor
```

Comment:
this approach makes perfect sense and conveys exactly the semantics we discussed, but there is something that doesn't feel right to me, for the following reasons:

- I feel that `styleFallback` and `defaultStyle` should have  the `lineColor` property, even if as `null`/`undefined` (not sure whether it would work), as that property is part of the style API. if the default value is `node['note']` even better because it conveys the semantics of what we want to achieve

- to me the logic in this line feels like it should live where we update all other styles, and in this part of the code we should simply read the property that we computed at "update time" (I see this logic more as BL than rendering logic, mostly to keep in line with the rest of the settings updating logic. I understand this is a personal preference, so I am not too fussed about it)

Just looking again at the code, I realized we are using the same logic a few lines above. This to me confirms that this check shouldn't live here but be part of the update logic.

thoughts?

Reply:
See latest commit: I think you are right that it cleans things up and moves logic to a better place. Not sure if the change to updateStyle is the best way to do it though, let me know if you see a better way.